packages/@react-aria/utils/src/shadowdom/DOMFunctions.ts

```diff
@@ -0,0 +1,85 @@
+// Source: https://github.com/microsoft/tabster/blob/a89fc5d7e332d48f68d03b1ca6e344489d1c3898/src/Shadowdomize/DOMFunctions.ts#L16
+
+export function nodeContains(
+  node: Node | null | undefined,
+  otherNode: Node | null | undefined
+): boolean {
+  if (!node || !otherNode) {
+    return false;
+  }
+
+  let currentNode: HTMLElement | Node | null | undefined = otherNode;
+
+  while (currentNode) {
+    if (currentNode === node) {
+      return true;
+    }
+
+    if (
+          typeof (currentNode as HTMLSlotElement).assignedElements !==
```

Comment:
can we format this a bit nicer?

Reply:
also maybe better to check `currentNode.tagName !== 'SLOT'` instead of looking for this method?